src/backend/jitsi_magnify/core/api.py

```diff
@@ -32,42 +37,63 @@ def exception_handler(exc, context):
 
     return drf_exception_handler(exc, context)
 
-class CustomTokenSerializer(TokenObtainPairSerializer):
-    """Api viewset to generate a custom token"""
 
-    @classmethod
-    def get_token(cls, user, room):
-        """Method to get the token created based on a user and a room which they wish to access"""
-        token = super().get_token(user)
+def create_payload(user, room, expiration):
+    """Create the payload so that it contains each information jitsi requires"""
+    token_payload = {
+        "user_id": user.id,
+        "exp": expiration,
+        "iat": datetime.datetime.utcnow(),
+        "moderator": user.is_staff,
+        "aud": "jitsi",
+        "iss": settings.JWT_CONFIGURATION.get("jitsi_app_id"),
+        "sub": "meet.jitsi",
+        "room": room,
+    }
+
+    jitsi_user = {
+        "avatar": settings.JWT_CONFIGURATION.get("guest_avatar"),
+        "name": user.username,
+        "email": user.email,
+    }
+
+    token_payload["context"] = {"user": jitsi_user}
 
-        # Add custom claims
-        jitsiUser = {'avatar': settings.JWT_CONFIGURATION.get("guest_avatar"),
-                'name': user.username,
-                'email': user.email}
+    return token_payload
 
-        token['context'] = {'user': jitsiUser}
 
-        token['moderator'] = user.is_staff
-        token['aud'] = "jitsi"
-        token['iss'] = settings.JWT_CONFIGURATION.get("jitsi_app_id")
-        token['sub'] = "meet.jitsi"
-        token['room'] = room
+def generate_token(user, room, expiration):
+    """Generate the access token that will give access to the room"""
+    token_payload = create_payload(user, room, expiration=expiration)
+    token = jwt.encode(
+        token_payload,
+        settings.JWT_CONFIGURATION.get("jitsi_secret_key"),
+        algorithm="HS256",
+    )
 
-        return token
+    return token
 
-class CustomTokenView(TokenObtainPairView):
-    """View for token creation"""
-    serializer_class = CustomTokenSerializer
 
-def obtainTokenView(request, room):
+@api_view(("GET",))
+def get_room_token_view(request, room):
     """View for token fetching based on a guest user and a specific room
     This redirects to jitsi with the token"""
     user, created = User.objects.get_or_create(username="guest")
```

Comment:
I think we should not create a user in database for a guest. This code made sense in joanie because al users are connected. Here, you can check in `generate_token` if `request.user.is_authenticated()` and use a generic user id, username and no email?